src/helpers/readme-helpers-trait.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * The readme.txt helpers trait file.
+ *
+ * @since   1.1.0
+ *
+ * @package Theme_Sniffer\Helpers
+ */
+
+declare( strict_types=1 );
+
+namespace Theme_Sniffer\Helpers;
+
+/**
+ * Sniffer helpers trait
+ *
+ * This trait contains some helper methods.
+ *
+ * @since 1.1.0
+ */
+trait Readme_Helpers {
+
+	/**
+	 * License data from imported from json.
+	 *
+	 * @var array $license_data
+	 *
+	 * @since 1.1.0
+	 */
+	protected $license_data;
+
+	/**
+	 * Initialize class and set class properties.
+	 *
+	 * @since 1.1.0
+	 */
+	public function init() {
+		$this->license_data = $this->set_license_data();
+	}
+
+	/**
+	 * Gets license validation data.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return array License validation data.
+	 */
+	public function set_license_data() : array {
+		$file = WP_PLUGIN_DIR . plugin_dir_path( '/theme-sniffer/assets/build/licenses.json' );
+		$file = file_get_contents( $file . 'licenses.json' );
+		return json_decode( $file, true );
+	}
+
+	/**
+	 * Find license matches.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param string $id License identififier.
+	 *
+	 * @return object $reponse Object containing license critera and match information.
+	 */
+	public function find_license( $id ) {
+		$response             = (object) [];
+		$response->provided   = $id;
+		$response->deprecated = [];
+		$response->live       = [];
+
+		// SPDX ID exact match, so skip loop.
+		$found = $this->license_data[ $response->provided ] ?? [];
+
+		// Check if license is deprecated.  Set data.
+		if ( $found ) {
```

Comment:
This and the next conditional can be merged to one `if/else`, no?

Reply:
doh yeah, I don't know what I was thinking there :laughing: